internal/commands/open_test.go

```diff
@@ -0,0 +1,144 @@
+package commands_test
+
+import (
+	"context"
+	"fmt"
+	"hellper/internal/bot"
+	"hellper/internal/commands"
+	filestorage "hellper/internal/file_storage"
+	"hellper/internal/log"
+	"hellper/internal/model"
+	"testing"
+
+	"github.com/slack-go/slack"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type openCommandFixture struct {
+	testName             string
+	expectError          bool
+	errorMessage         string
+	ctx                  context.Context
+	mockLogger           log.Logger
+	mockClient           bot.Client
+	mockRepository       model.Repository
+	mockFilestorage      filestorage.Driver
+	triggerID            string
+	mockDialogSubmission bot.DialogSubmission
+}
+
+func (f *openCommandFixture) setup(t *testing.T) {
+	var (
+		loggerMock      = log.NewLoggerMock()
+		clientMock      = bot.NewClientMock()
+		repositoryMock  = model.NewRepositoryMock()
+		filestorageMock = filestorage.NewFileStorageMock()
+	)
+
+	f.ctx = context.Background()
+	f.mockLogger = loggerMock
+	f.mockClient = clientMock
+	f.mockRepository = repositoryMock
+	f.mockFilestorage = filestorageMock
+
+	loggerMock.On("Info", f.ctx, mock.AnythingOfType("string"), mock.AnythingOfType("[]log.Value")).Return()
+	clientMock.On("OpenDialog", f.triggerID, mock.AnythingOfType("slack.Dialog")).Return(nil)
+	clientMock.On("AddPin", mock.AnythingOfType("string"), mock.AnythingOfType("slack.ItemRef")).Return(nil)
+	clientMock.On("PostMessage", mock.AnythingOfType("string"), mock.AnythingOfType("[]slack.MsgOption")).Return("", "", nil)
+	clientMock.On("JoinConversationContext", f.ctx, mock.AnythingOfType("string")).Return(new(slack.Channel), "", []string{}, nil)
+	clientMock.On("InviteUsersToConversationContext", f.ctx, mock.AnythingOfType("string"), mock.AnythingOfType("[]string")).Return(new(slack.Channel), nil)
+	clientMock.On("SetTopicOfConversation", mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return(new(slack.Channel), nil)
+	clientMock.On("GetUserInfoContext", f.ctx, mock.AnythingOfType("string")).Return(new(slack.User), nil)
+	clientMock.On("CreateConversationContext", f.ctx, mock.AnythingOfType("string"), mock.AnythingOfType("bool")).Return(new(slack.Channel), nil)
+	repositoryMock.On("InsertIncident", mock.AnythingOfType("*model.Incident")).Return(int64(1), nil)
+	repositoryMock.On("AddPostMortemUrl", mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return(nil)
+	filestorageMock.On("CreatePostMortemDocument", f.ctx, mock.AnythingOfType("string")).Return(string(""), nil)
+}
+
+func TestOpenIncidentDialog(t *testing.T) {
+	var f openCommandFixture
+	t.Run("Dialog created properly", func(t *testing.T) {
+		f.setup(t)
+		err := commands.OpenStartIncidentDialog(f.mockClient, f.triggerID)
```

Comment:
This shows that maybe our code is wrong. You a sending an empty string and no error returns.

Reply:
I realized this when building the test :/